python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -252,6 +251,42 @@ public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functi
     return Optional.empty();
   }
 
+  public static Optional<FunctionSymbol> getFunctionSymbol(@Nullable Symbol symbol) {
+    if (symbol == null) {
+      return Optional.empty();
+    }
+
+    if (symbol.is(Symbol.Kind.FUNCTION)) {
+      return Optional.of(symbol).map(FunctionSymbol.class::cast);
+    }
+
+    if (symbol.is(Symbol.Kind.AMBIGUOUS)) {
+      var ambiguousSymbol = (AmbiguousSymbol) symbol;
+
+      var functionSymbols = ambiguousSymbol
+        .alternatives()
+        .stream()
+        .filter(alternative -> alternative.is(Symbol.Kind.FUNCTION))
```

Comment:
instead of filtering `FUNCTION`, we need to make sure that every alternative is a `FUNCTION`